core/src/main/scala/kafka/coordinator/group/GroupMetadata.scala

```diff
@@ -649,26 +650,27 @@ private[group] class GroupMetadata(val groupId: String, initialState: GroupState
     }
   }
 
-  def failPendingOffsetWrite(topicPartition: TopicPartition, offset: OffsetAndMetadata): Unit = {
+  def failPendingOffsetWrite(topicIdPartition: TopicIdPartition, offset: OffsetAndMetadata): Unit = {
+    val topicPartition = topicIdPartition.topicPartition
     pendingOffsetCommits.get(topicPartition) match {
       case Some(pendingOffset) if offset == pendingOffset => pendingOffsetCommits.remove(topicPartition)
       case _ =>
     }
   }
 
-  def prepareOffsetCommit(offsets: Map[TopicPartition, OffsetAndMetadata]): Unit = {
+  def prepareOffsetCommit(offsets: Map[TopicIdPartition, OffsetAndMetadata]): Unit = {
     receivedConsumerOffsetCommits = true
-    pendingOffsetCommits ++= offsets
+    offsets.foreach { case (k,v) => pendingOffsetCommits += k.topicPartition -> v }
```

Comment:
nit: We could use `forKeyValue` here. Moreover, let's use `(topicIdPartition, offsetAndMetadata)` (note the space after the `,`). instead of `(k, v)`.